pypuppetdb/types.py

```diff
@@ -128,22 +176,36 @@ def __str__(self):
     def __unicode__(self):
         return self.__string
 
+    def events(self):
+        """Get all events for this report."""
+        return self.__api.events(query=self.__query_scope)
```

Comment:
I also have a lot of work done in Puppetboard including pagination (for reports for nowbut can be expanded), and environment awareness.